### PR TITLE
WIP: Kubernetes client for ST

### PR DESCRIPTION
**Pull request by serrss**
###This PR still in progress

### Type of change
- Refactoring

### Description
After long execution ST we decided that using fabric8 client to emulate ST will be a more stable solution. So it is the main reason for this refactoring.
This PR has implementation fabric8 Kubernetes client for ST. Kubernetes client has all usual methods to create/update/delete all Strizmi entities.
Also, in scope of this PR, all helped methods like `wait` were moved to `StUtils` class to split client and other logic. 

TODO:
- [ ] Remove usage CMD kube client in the remaining system tests
- [ ] Order methods in kude client 
- [ ] Deprecate old methods
- [ ] Add JavaDocs
- [ ] Make sure that Kubernetes works fine without additional configuration
- [ ] Fix check-style issues

### Checklist
- [ ] Update/write design documentation in `./design`
- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: test/src/main/java/io/strimzi/test/k8s/Kubernetes.java [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+    }
+
+    /**
+     * Gets pod
+     */
+    public Pod getPod(String name) {
+        return CLIENT.inNamespace(namespace).pods().withName(name).get();
+    }
+
+    public Date getPodCreateTimestamp(String podName) {
```
Comment (Frawless) [inline test/src/main/java/io/strimzi/test/k8s/Kubernetes.java]: `getPodCreationTimestamp()` ? I am not an English master but it make better sense to me with this name.

File: test/src/main/java/io/strimzi/test/k8s/Kubernetes.java [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+    public boolean getServiceStatus (String serviceName) {
+        return CLIENT.inNamespace(namespace).services().withName(serviceName).isReady();
+    }
+
+    public void deleteService (String serviceName) {
+        CLIENT.inNamespace(namespace).services().withName(serviceName).delete();
+    }
+
+    public Job createJob(Job job) {
+        return CLIENT.inNamespace(namespace).extensions().jobs().create(job);
```
Comment (Frawless) [inline test/src/main/java/io/strimzi/test/k8s/Kubernetes.java]: `jobs()` in this context is deprecated, maybe find another solution which will be supported in future versions?

File: test/src/main/java/io/strimzi/test/k8s/OpenShift.java [hunk 3]
```diff
@@ -22,7 +22,7 @@ public boolean isAvailable() {
     @Override
     public boolean isClusterUp() {
         try {
-            Exec.exec(OC, "cluster", "status");
+//            Exec.exec(OC, "cluster", "status");
```
Comment (Frawless) [inline test/src/main/java/io/strimzi/test/k8s/OpenShift.java]: why dd you remove this line?

File: test/src/main/java/io/strimzi/test/k8s/OpenShift.java [hunk 4]
```diff
@@ -31,7 +31,7 @@ public boolean isClusterUp() {
                     // In this case it is still coming up, so wait for rather than saying it's not up
                     TestUtils.waitFor("oc cluster up", 1_000, 60_000, () -> {
                         try {
-                            Exec.exec(OC, "cluster", "status");
```
Comment (Frawless) [inline test/src/main/java/io/strimzi/test/k8s/OpenShift.java]: same as above

File: test/src/main/java/io/strimzi/test/k8s/Kubernetes.java [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+
+    public String getNamespace() {
+        return this.namespace;
+    }
+
+    public void deleteNamespace(String name) {
+        CLIENT.namespaces().withName(name).delete();
+    }
+
+    public void deleteConfigMap (String configManName) {
```
Comment (Frawless) [inline test/src/main/java/io/strimzi/test/k8s/Kubernetes.java]: typo? -> `configMapName`

File: test/src/main/java/io/strimzi/test/k8s/Kubernetes.java [hunk 6]
```diff
@@ -0,0 +1,314 @@
...
+        CLIENT.inNamespace(namespace).configMaps().withName(configManName).delete();
+    }
+
+    public boolean getConfigMapStatus (String configManName) {
```
Comment (Frawless) [inline test/src/main/java/io/strimzi/test/k8s/Kubernetes.java]: same as above

File: test/src/main/java/io/strimzi/test/k8s/Kubernetes.java [hunk 7]
```diff
@@ -0,0 +1,314 @@
...
+            LOGGER.warn("Exception running command {} on pod: {}", command, e.getMessage());
+            return "";
+        }
+    }
+
+    public List<Pod> listPods(LabelSelector selector) {
+        return CLIENT.inNamespace(namespace).pods().withLabelSelector(selector).list().getItems();
+    }
+
+    public List<Pod> listPods(Map<String, String> labelSelector) {
```
Comment (Frawless) [inline test/src/main/java/io/strimzi/test/k8s/Kubernetes.java]: You are using `Collections.singletonMap()` everywhere. From my POV it's better solution to overload this method again with 2 string parameters and create singletonMap inside overloaded method

**Comment by serrss** [comment]
@Frawless Yeah, thanks for the comments.
I have a plan to rename `BaseKubeClient` to `BaseCMDKubeClient`. 
As for the suggestion to implement KubeClient we need to think more deeply, because, maybe a better solution will be having an interface for fabric8 clients. 
As for `listing sercrets/pods/resources`, I removed methods arguments like a namespace in X methods because of redundant using this argument. Right now we have only one ST where we need to manage namespace during one ST, so we no need to care about namespace in many cases. But, if you need to do it in the scope of one ST, I guess we need to have DSL for `namespace`(I will add it in TODO list).
Thanks, for a good point for Fabric8 wait process!

**Comment by serrss** [comment]
PR was closed because of new one #1544

**Pull request closed without merging**